internal/build/phase_config_provider.go

```diff
@@ -122,14 +123,38 @@ func WithBinds(binds ...string) PhaseConfigProviderOperation {
 	}
 }
 
-func WithDaemonAccess() PhaseConfigProviderOperation {
+func WithDaemonAccess(dockerHost string) PhaseConfigProviderOperation {
 	return func(provider *PhaseConfigProvider) {
 		WithRoot()(provider)
-		bind := "/var/run/docker.sock:/var/run/docker.sock"
-		if provider.os == "windows" {
-			bind = `\\.\pipe\docker_engine:\\.\pipe\docker_engine`
+		if dockerHost == "inherit" {
+			dh, ok := os.LookupEnv("DOCKER_HOST")
+			if ok {
+				dockerHost = dh
+			} else {
+				dockerHost = ""
+			}
+		}
+		var bind string
+		if dockerHost == "" {
+			bind = "/var/run/docker.sock:/var/run/docker.sock"
+			if provider.os == "windows" {
+				bind = `\\.\pipe\docker_engine:\\.\pipe\docker_engine`
+			}
+		} else {
+			switch {
+			case strings.HasPrefix(dockerHost, "tcp://"):
+				provider.ctrConf.Env = append(provider.ctrConf.Env, fmt.Sprintf(`DOCKER_HOST=%s`, dockerHost))
+			case strings.HasPrefix(dockerHost, "unix://"):
+				bind = fmt.Sprintf("%s:/var/run/docker.sock", strings.TrimPrefix(dockerHost, "unix://"))
+			case strings.HasPrefix(dockerHost, "npipe://"):
+				bind = fmt.Sprintf(`%s:\\.\pipe\docker_engine`, strings.TrimPrefix(dockerHost, "npipe://"))
```

Comment:
As you mentioned, it would be good to see tests for these cases as well, though not necessarily in all 3 phases that call it (`Analyze`, `Export` and `Creater`), I'd be happy to just see it in `Creator`